Sources/SwiftProtobuf/TextEncoder.swift

```diff
@@ -0,0 +1,168 @@
+// ProtobufRuntime/Sources/Protobuf/ProtobufTextEncoding.swift - Text format encoding support
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+// -----------------------------------------------------------------------------
+///
+/// Text format serialization engine.
+///
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+public class TextEncoder {
+    var text: [String] = []
+    private var tabLevel = 0
+
+    init() {}
+
+    var result: String { return text.joined(separator: "") }
+
+    func append(text newText: String) {
```

Comment:
Everything here is 'internal' (used only in the library) except for the `TextEncoder` class itself.  I've refactored slightly to eliminate this one redundant method.